api/src/pcapi/core/providers/titelive_product_add_gtl_jsondata.py

```diff
@@ -0,0 +1,137 @@
+                ean = row[COLUMN_INDICES["EAN"]]
+                gtl_id = row[COLUMN_INDICES["CODE_GTL"]]
+                code_clil = row[COLUMN_INDICES["CODE_CLIL"]]
+                csr_id = row[COLUMN_INDICES["CODE_CSR"]]
+                csr = get_csr_from_csr_id(csr_id)
+
+                # Using offer_models.OfferExtraData is much slower
+                extra_data = {
+                    EXTRA_DATA_KEYS["ean"]: ean,
+                    EXTRA_DATA_KEYS["code_gtl"]: gtl_id,
+                    EXTRA_DATA_KEYS["code_clil"]: code_clil,
+                    EXTRA_DATA_KEYS["code_csr"]: csr_id,
+                    EXTRA_DATA_KEYS["rayon"]: csr["label"] if csr else None,
+                }
+
+                if not ean in ean_datas:
+                    ean_datas[ean] = extra_data  # this is useful for performance reasons
+                    ean_data_list.append(extra_data)
+                else:
+                    ean_duplicates.append(ean)
+
+        eans = set(ean_datas.keys())
+
+        products = offers_models.Product.query.filter(offers_models.Product.idAtProviders.in_(eans)).all()
+
+        logger.info("Total rows: %s", total_rows)
+        logger.info("Total EANs: %s", len(eans))
+        logger.info("Ignoring %s EANs present more than once in CSV extract: %s", len(ean_duplicates), ean_duplicates)
+        logger.info("Total matching products in database: %s", len(products))
+
+        products_and_eans_diff_count = len(eans) - len(products)
+        print(f"{products_and_eans_diff_count} EANs in CSV are not present in the database, aborting...")
+        if not settings.IS_DEV and products_and_eans_diff_count != 0:
+            raise CsvGtlImportException(
+                f"{products_and_eans_diff_count} EANs in CSV are not present in the database, aborting..."
+            )
+
+        if settings.IS_DEV and products_and_eans_diff_count != 0:
+            logger.info(
+                "%s EANs in CSV are not present in the database, ignoring as in dev environment...",
+                products_and_eans_diff_count,
+            )
+
+        logger.info("It will now update %s entries", len(products))
+
+        for product in products:
+            product.extraData[EXTRA_DATA_KEYS["ean"]] = extra_data[EXTRA_DATA_KEYS["ean"]]
+            product.extraData[EXTRA_DATA_KEYS["code_gtl"]] = extra_data[EXTRA_DATA_KEYS["code_gtl"]]
+            product.extraData[EXTRA_DATA_KEYS["code_clil"]] = extra_data[EXTRA_DATA_KEYS["code_clil"]]
+            product.extraData[EXTRA_DATA_KEYS["code_csr"]] = extra_data[EXTRA_DATA_KEYS["code_csr"]]
+            product.extraData[EXTRA_DATA_KEYS["rayon"]] = (
+                extra_data[EXTRA_DATA_KEYS["rayon"]] or product.extraData[EXTRA_DATA_KEYS["rayon"]]
+            )
+            if not dry:
+                db.session.add(product)
+
+        if dry:
+            logger.info("[dryrun] It will update %s products (Run without --dry-run option to apply change to database)", len(products))
```

Comment:
```suggestion
            logger.info("[dryrun] It would update %s products (Run without --dry-run option to apply changes to database)", len(products))
```